### ISSUE TITLE
Converting LLA to UPS returns NaN coordinates for the poles

### ISSUE THREAD

**Issue by evanfields** [JuliaGeo/Geodesy.jl#57]
Converting `LLA` to `UTMZ` returns a `UTMZ` with `NaN` coordinates for lat-lons at the poles:
```
julia> converter = UTMZfromLLA(wgs84)
UTMZfromLLA(WGS84)

julia> converter(LLA(90.0, 0.0, 0.0))
UTMZ(NaN, NaN, 0.0, zone=polar (north))

julia> converter(LLA(-90.0, 0.0, 0.0))
UTMZ(NaN, NaN, 0.0, zone=polar (south))
```
I know very little about geodesy so thought maybe this was intentional? But then it's surprising that there is a `UTMZ` that converts back to the poles:
```
julia> LLAfromUTMZ(wgs84)(UTMZ(2e6, 2e6, 0, true)) == LLA(90.0, 180.0, 0.0)
true
```

**Comment by c42f** [comment]
Well UTM zones aren't usually used at the poles — for that you should use the polar stereographic projection instead (UPS). GeographicLib https://geographiclib.sourceforge.io/cgi-bin/GeoConvert just gives an error for high latitudes, so perhaps we should do so in the UTMZ constructor.

Having said that, the underlying Transverse Mercator projection should be defined at the poles. So it seems that the choices here are to make this work by fixing the projection code, or validate the latitude in the UTMZ constructor.

**Comment by evanfields** [comment]
Interesting. I knew the narrow UTM bands didn't work so well at the poles, but I interpreted `zone=polar (north)` as Geodesy automatically doing UPS at the poles. I guess not? But then what is zone 0?

**Comment by c42f** [comment]
> I knew the narrow UTM bands didn't work so well at the poles

Yes, I think it's more a matter of convention (the desire not to have really thin zones) than any mathematical problem. The singularities of Transverse Mercator lie on the Equator.

> I interpreted zone=polar (north) as Geodesy automatically doing UPS at the poles

Oh yes, very sorry - I clearly didn't read your example carefully enough (and I haven't used this package for a while)! In that case you're quite correct - this is just a bug.

**Comment by c42f** [comment]
I'm not sure I have bandwidth to look into this. But if you're willing to take a look, I suggest comparing our UPS transformation with that of GeographicLib here https://sourceforge.net/p/geographiclib/code/ci/release/tree/src/PolarStereographic.cpp. That might help uncover what's wrong.

**Comment by c42f** [comment]
Fix should be in release 1.0.1 https://github.com/JuliaRegistries/General/pull/27079
